tg_bot.py

```diff
@@ -40,9 +41,17 @@ def display_tweet(self, context):
         print('starting display parameter')
         job = context.job
         global KILL_SIGNAL
+        previous_date = datetime.now() - timedelta(days=1)
         tid = pd.read_csv(self.dirpath + 'tid_store.csv')
-        for index, tid_store in tid.iterrows():
-            
+        tid["Date_time"]= pd.to_datetime(tid["Date_time"])
+
+        for index, tid_store in tid[::-1].iterrows():
+            scrape_date = tid_store['Date_time']
+            if scrape_date<=previous_date:
+                context.bot.send_message(job.context, text = "Rest are the tweets which were scraped before")
+                break
+            else:
```

Comment:
I'm still not clear of how you are handling the data if there is no previous data to compare with? For instance when the bot has been just started and there is no previous day? You just continue what?

and what if we have a missing date?

Reply:
When the bot is just being started, it will just continue to send all the tweets to the user like the present case.